website/source/docs/provisioners/chef.html.markdown

```diff
@@ -0,0 +1,96 @@
+---
+layout: "docs"
+page_title: "Provisioner: chef"
+sidebar_current: "docs-provisioners-chef"
+description: |-
+  The `chef` provisioner invokes a Chef Client run on a remote resource after first installing and configuring Chef Client on the remote resource. The `chef` provisioner supports both `ssh` and `winrm` type connections.
+---
+
+# Chef Provisioner
+
+The `chef` provisioner invokes a Chef Client run on a remote resource after first installing
+and configuring Chef Client on the remote resource. The `chef` provisioner supports both `ssh`
+and `winrm` type [connections](/docs/provisioners/connection.html).
+
+## Requirements
+
+In order for the `chef` provisioner to work properly, you need either `cURL` (when using
+a `ssh` type connection) or `PowerShell 2.0` (when using a `winrm` type connection) to be
+available on the target machine.
+
+## Example usage
+
+```
+# Start a initial chef run on a resource
+resource "aws_instance" "web" {
+    ...
+    provisioner "chef"  {
+        attributes {
+            "key" = "value"
+            "app" {
+                "cluster1" {
+                    "nodes" = ["webserver1", webserver2]
+                }
+            }
+        }
+        environment = "_default"
+        run_list = ["cookbook::recipe"]
+        node_name = "webserver1"
+        server_url = "https://chef.company.com/organizations/org1"
+        validation_client_name = "chef-validator"
+        validation_key_path = "../chef-validator.pem"
+        version = "11.18.6"
+    }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `attributes (hash)` - (Optional) A hash with initial node attributes for the new node.
+  See example.
```

Comment:
In the [Terraform config language](https://terraform.io/docs/configuration/syntax.html) we've been calling these "maps" rather than "hashes".